zerver/lib/notification_data.py

```diff
@@ -176,15 +176,21 @@ def get_email_notification_trigger(self, acting_user_id: int, idle: bool) -> Opt
 
 def user_allows_notifications_in_StreamTopic(
     stream_is_muted: bool,
-    topic_is_muted: bool,
+    visibility_policy: int,
     stream_specific_setting: Optional[bool],
     global_setting: bool,
 ) -> bool:
     """
-    Captures the hierarchy of notification settings, where muting is considered first, followed
-    by stream-specific settings, and the global-setting in the UserProfile is the fallback.
+    Captures the hierarchy of notification settings, where visibility policy is considered first,
+    followed by stream-specific settings, and the global-setting in the UserProfile is the fallback.
     """
-    if stream_is_muted or topic_is_muted:
+    topic_is_muted = visibility_policy == UserTopic.MUTED
+    topic_is_unmuted = visibility_policy == UserTopic.UNMUTED
+
+    if stream_is_muted and not topic_is_unmuted:
+        return False
+
```

Comment:
IMO we should probably have the `visibility_policy == UserTopic.MUTED` as part of the `if` statement itself, instead of creating separate `bool`.

Since `topic_is_muted` and `topic_is_unmuted` are not mutually exclusive, to a new reader it might be confusing (as in, shouldn't it be that `topic_is_muted == !topic_is_unmuted`?). Explicitly using the values from the model would prompt the reader to go to the model to read the definitions of `UNMUTED` etc, where we have comments explaining what these mean.

Reply:
More broadly, I think it would be nice to avoid using the terms "muted"/"unmuted" anywhere in the code. `visibility_policy = UserTopic.UNMUTED` is much more explicit.
(For this reason, it is better to have a single function (which we have now) instead of having functions like `do_mute_topic`/`do_unmute_topic` etc)

Reply:
Makes sense -- Updated the PR.